packages/cfd/src/Containers/cfd-password-modal/cfd-password-form/cfd-password-form.tsx

```diff
@@ -0,0 +1,297 @@
+import React from 'react';
+import { Formik, FormikErrors } from 'formik';
+import { getLegalEntityName, getErrorMessages, getCFDPlatformLabel, CFD_PLATFORMS, CFD_text } from '@deriv/shared';
+import { Text, FormSubmitButton, PasswordMeter, PasswordInput } from '@deriv/components';
+import { Localize } from '@deriv/translations';
+import {
+    getDxCompanies,
+    getFormattedJurisdictionCode,
+    TDxCompanies,
+} from '../../../Stores/Modules/CFD/Helpers/cfd-config';
+import { TCFDPasswordFormReusedProps, TCFDPasswordFormValues, TOnSubmitPassword } from '../../props.types';
+import { ACCOUNT_CATEGORY, PASSWORD_ERRORS } from '../../../Constants/cfd-password-modal-constants';
+import PasswordStep from './password-step';
+import { observer, useStore } from '@deriv/stores';
+
+type TCFDPasswordFormProps = TCFDPasswordFormReusedProps & {
+    account_title: string;
+    account_type: {
+        category?: string;
+        type?: string;
+    };
+    closeModal: () => void;
+    error_type?: string;
+    form_error?: string;
+    has_mt5_account: boolean;
+    is_bvi: boolean;
+    is_dxtrade_allowed: boolean;
+    is_real_financial_stp: boolean;
+    jurisdiction_selected_shortcode: string;
+    onCancel: () => void;
+    onForgotPassword: () => void;
+    should_set_trading_password: boolean;
+    show_eu_related_content: boolean;
+    submitPassword: TOnSubmitPassword;
+};
+
+const getAccountTitle = (
+    platform: string,
+    account_type: {
+        category?: string;
+        type?: string;
+    },
+    account_title: string
+) => {
+    if (platform === CFD_PLATFORMS.DXTRADE) {
+        return getDxCompanies()[account_type.category as keyof TDxCompanies][
+            account_type.type as keyof TDxCompanies['demo' | 'real']
+        ].short_title;
+    }
+
+    return account_title;
+};
+
+const handlePasswordInputChange = (
```

Comment:
Why are we prop drilling this function ?
This can be extracted out and imported into the files.